packages/nimble-components/src/text-area/index.ts

```diff
@@ -26,6 +27,99 @@ export class TextArea extends FoundationTextArea {
      */
     @attr
     public appearance: TextAreaAppearance = TextAreaAppearance.outline;
+
+    /**
+     * A message explaining why the value is invalid.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: error-text
+     */
+    @attr({ attribute: 'error-text' })
+    public errorText?: string;
+
+    /**
+     * Whether to display the error state.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: error-visible
+     */
+    @attr({ attribute: 'error-visible', mode: 'boolean' })
+    public errorVisible = false;
+
+    /**
+     * The width of the vertical scrollbar, if displayed.
+     * @internal
+     */
+    @observable
+    public scrollbarWidth = 0;
+
+    private resizeObserver?: ResizeObserver;
+    private updateScrollbarWidthQueued = false;
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.resizeObserver = new ResizeObserver(this.onResize);
+        this.resizeObserver.observe(this);
+    }
+
+    /**
+     * @internal
+     */
+    public override disconnectedCallback(): void {
+        this.resizeObserver?.disconnect();
+    }
+
+    /**
+     * @internal
+     */
+    public onTextAreaInput(): void {
+        this.handleTextInput();
+        this.queueUpdateScrollbarWidth();
+    }
+
+    // If a property can affect whether a scrollbar is visible, we need to
+    // call queueUpdateScrollbarWidth() when it changes. The exceptions are
+    // properties that affect size (e.g. height, width, cols, rows), because
+    // we already have a ResizeObserver handling those changes.
+
+    /**
+     * @internal
+     */
+    public placeholderChanged(): void {
```

Comment:
Need one for errorVisibleChanged

Reply:
Now that we are always reserving space for the icon, errorVisible changing won't cause a scrollbar to appear/disappear.

Reply:
Makes sense. Please call that out in the exceptions list above with the reasoning.

Reply:
Done